livre/lib/data/BoundingAxis.cpp

```diff
@@ -0,0 +1,368 @@
+    const float dataToWorldScale = worldSpaceLenght / rangeLenght;
+    const float startTickValue = bbAxisData.tickSize * std::ceil( range.x() / bbAxisData.tickSize );
+    const float worldSpaceOffset = ( startTickValue - range.x( )) * dataToWorldScale;
+    const float worldSpaceTickSize = bbAxisData.tickSize * dataToWorldScale;
+
+    const int32_t numberOfTick = ( range.y() - startTickValue ) / bbAxisData.tickSize + 1;
+    const Vector3f axisDir = vmml::normalize( end - start );
+
+    Vector3f cross =  vmml::normalize( vmml::cross( axisDir, normal ));
+    size_t tickAxis;
+    if( cross.find_max() <= std::numeric_limits<float>::min( ))
+        tickAxis = cross.find_min_index( );
+    else
+        tickAxis = cross.find_max_index( );
+
+    for( int32_t i = 0; i < numberOfTick; ++i )
+    {
+        float tickHeight = 0.014f;
+        Vector4f color( 1.0, 0.69, 0.69, 1.0 );
+        const int32_t startTickValueInteger = std::round( startTickValue / bbAxisData.factor);
+        const int32_t tickSizeInteger = std::round( bbAxisData.tickSize / bbAxisData.factor );
+
+        if((( startTickValueInteger + tickSizeInteger * i ) % ( tickSizeInteger * 5 )) == 0 )
+        {
+            tickHeight *= 1.5;
+            color = Vector4f( 1.0, 1.0, 0.69, 1.0 );
+        }
+
+        int sign;
+        if( flipTick )
+            sign = -(( 0.0f < cross[ tickAxis ]) - ( cross[ tickAxis ] < 0.0f ));
+        else
+            sign = ( 0.0f < cross[ tickAxis ]) - ( cross[ tickAxis ] < 0.0f );
+
+        Vector3f posOnAxis = start + axisDir * ( worldSpaceOffset + worldSpaceTickSize * i );
+        _append3dVector( bbAxisData.vertices, posOnAxis );
+        posOnAxis[ tickAxis ] = posOnAxis[ tickAxis ] + sign * tickHeight;
+        _append3dVector( bbAxisData.vertices, posOnAxis );
+
+        for( uint32_t j = 0; j < 2; ++j )
+        {
+            _append3dVector( bbAxisData.normals, normal );
+            _append3dVector( bbAxisData.normals2, normal2 );
+            _append4dVector( bbAxisData.colors, color );
+            bbAxisData.types.push_back( AXIS );
+        }
+
+        _nVertices += 2;
+    }
+}
+
+const Vector2f BoundingAxis::_computeRange( const VolumeInformation& volInfo,
+                                             const size_t axis ) const
+{
+    if( volInfo.resolution.x() == -1 )
+        return Vector2f( 0.0f, volInfo.worldSize[ axis ]);
+    else
+        return Vector2f( 0.0f, volInfo.voxels[ axis ] * volInfo.resolution[ axis ]);
+}
+
+void BoundingAxis::_append3dVector( std::vector< float >& container, const Vector3f& vector )
+{
+    container.push_back( vector.x( ));
+    container.push_back( vector.y( ));
+    container.push_back( vector.z( ));
+}
+
+void BoundingAxis::_append4dVector( std::vector< float >& container, const Vector4f& vector )
+{
+    container.push_back( vector.x( ));
+    container.push_back( vector.y( ));
+    container.push_back( vector.z( ));
+    container.push_back( vector.w( ));
```

Comment:
It's not that it can't.
When a "member" function doesn't need any state of the object to which it belongs it's better to make it a free function to keep the class interface clean. In this case the member function is even private, so making it a free function contained in the .cpp allows you to change it in the future without touching the .h file.